tv_apps/fling-player/js/fling_player.js

```diff
@@ -100,14 +102,33 @@
 
   // UI handling
 
-  proto.setLoading = function fp_setLoading(loading) {
+  proto.showLoading = function (loading) {
     this._loadingUI.hidden = !loading;
   };
 
   /**
+   * @param {String} state 'playing' or 'paused'
+   */
+  proto.setPlayButtonState = function (state) {
+    switch (state) {
+      case 'playing':
+        this._playButton.textContent = 'Play';
+      break;
+
+      case 'paused':
+        this._playButton.textContent = 'Pause';
+      break;
+    }
+  };
```

Comment:
The text to indicate state seems wrong to me. Although it's just for your development testing, it's better to make it in a correct state. Please set: 
playing --> Pause
paused --> Play